browser/brave_stats/brave_stats_updater.cc

```diff
@@ -0,0 +1,375 @@
+}
+
+void BraveStatsUpdater::DisableThresholdPing() {
+  pref_service_->SetBoolean(kThresholdCheckMade, true);
+  pref_service_->ClearPref(kThresholdQuery);
+}
+
+void BraveStatsUpdater::QueueServerPing() {
+  if (IsReferralInitialized()) {
+    StartServerPingStartupTimer();
+  } else {
+    pref_change_registrar_.reset(new PrefChangeRegistrar());
+    pref_change_registrar_->Init(pref_service_);
+    pref_change_registrar_->Add(
+        kReferralInitialization,
+        base::Bind(&BraveStatsUpdater::OnReferralInitialization,
+                   base::Unretained(this)));
+  }
+}  // namespace brave
+
+void BraveStatsUpdater::OnReferralInitialization() {
+  StartServerPingStartupTimer();
+}
+
+void BraveStatsUpdater::StartServerPingStartupTimer() {
+  server_ping_startup_timer_->Start(
+      FROM_HERE, base::TimeDelta::FromSeconds(kUpdateServerStartupPingDelay),
+      this, &BraveStatsUpdater::OnServerPingTimerFired);
+  DCHECK(server_ping_startup_timer_->IsRunning());
+}
+
+net::NetworkTrafficAnnotationTag
+    BraveStatsUpdater::AnonymousStatsAnnotation() {
+  return net::DefineNetworkTrafficAnnotation("brave_stats_updater", R"(
+    semantics {
+      sender:
+        "Brave Stats Updater"
+      description:
+        "This service sends anonymous usage statistics to Brave."
+      trigger:
+        "Stats are automatically sent at intervals while Brave "
+        "is running."
+      data: "Anonymous usage statistics."
+      destination: WEBSITE
+    }
+    policy {
+      cookies_allowed: NO
+      setting:
+        "This feature cannot be disabled by settings."
+      policy_exception_justification:
+        "Not implemented."
+    })");
+}
+
+void BraveStatsUpdater::SendServerPing() {
+  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
+  auto traffic_annotation = AnonymousStatsAnnotation();
+  auto resource_request = std::make_unique<network::ResourceRequest>();
+  auto stats_updater_params =
+      std::make_unique<brave_stats::BraveStatsUpdaterParams>(pref_service_);
+
+  auto endpoint = BuildStatsEndpoint(kBraveUsageStandardPath);
+  resource_request->url = GetUpdateURL(endpoint, *stats_updater_params);
+  resource_request->load_flags =
+      net::LOAD_DO_NOT_SEND_COOKIES | net::LOAD_DO_NOT_SAVE_COOKIES |
+      net::LOAD_BYPASS_CACHE | net::LOAD_DISABLE_CACHE |
+      net::LOAD_DO_NOT_SEND_AUTH_DATA;
+  resource_request->headers.SetHeader(
+      "X-Brave-API-Key",
+      brave_stats::GetAPIKey());
+  network::mojom::URLLoaderFactory* loader_factory =
+      g_browser_process->system_network_context_manager()
+          ->GetURLLoaderFactory();
+  simple_url_loader_ = network::SimpleURLLoader::Create(
+      std::move(resource_request), traffic_annotation);
+  simple_url_loader_->DownloadHeadersOnly(
+      loader_factory,
+      base::BindOnce(&BraveStatsUpdater::OnSimpleLoaderComplete,
+                     base::Unretained(this), std::move(stats_updater_params)));
+}
+
+void BraveStatsUpdater::SendThresholdPing() {
```

Comment:
maybe this is `SendUserTriggeredPing`? Just want to have more understandable name